sonar-ballerina-plugin/src/main/java/io/ballerina/sonar/BallerinaSensor.java

```diff
@@ -0,0 +1,260 @@
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.ballerina.sonar.Constants.ANALYSIS_RESULTS_FILE_PATH;
+import static io.ballerina.sonar.Constants.BUG;
+import static io.ballerina.sonar.Constants.BUILT_IN;
+import static io.ballerina.sonar.Constants.EXTERNAL;
+import static io.ballerina.sonar.Constants.ISSUES_FILE_PATH;
+import static io.ballerina.sonar.Constants.ISSUE_END_LINE;
+import static io.ballerina.sonar.Constants.ISSUE_END_LINE_OFFSET;
+import static io.ballerina.sonar.Constants.ISSUE_FILE_PATH;
+import static io.ballerina.sonar.Constants.ISSUE_MESSAGE;
+import static io.ballerina.sonar.Constants.ISSUE_RULE_ID;
+import static io.ballerina.sonar.Constants.ISSUE_RULE_KIND;
+import static io.ballerina.sonar.Constants.ISSUE_SOURCE;
+import static io.ballerina.sonar.Constants.ISSUE_START_LINE;
+import static io.ballerina.sonar.Constants.ISSUE_START_LINE_OFFSET;
+import static io.ballerina.sonar.Constants.RULE_REPOSITORY_KEY;
+import static io.ballerina.sonar.Constants.SONAR_SCANNER_OFFSET;
+import static io.ballerina.sonar.Constants.VULNERABILITY;
+
+/**
+ * Represents the implementation of the {@link Sensor} class for Ballerina.
+ *
+ * @since 0.1.0
+ */
+class BallerinaSensor implements Sensor {
+    private final Logger logger = Loggers.get(BallerinaSensor.class);
+    private final BallerinaLanguage language;
+    private final ArrayList<String> externalRules = new ArrayList<>();
+
+    public BallerinaSensor(BallerinaLanguage language) {
+        this.language = language;
+    }
+
+    @Override
+    public void describe(SensorDescriptor descriptor) {
+        descriptor.onlyOnLanguage(language.getKey()).name(language.getName() + " Sensor");
+    }
+
+    @Override
+    public void execute(SensorContext sensorContext) {
+        FileSystem fileSystem = sensorContext.fileSystem();
+        FilePredicate mainFilePredicate = sensorContext.fileSystem().predicates()
+                .and(fileSystem.predicates().hasLanguage(language.getKey()),
+                        fileSystem.predicates().hasType(InputFile.Type.MAIN));
+        Map<String, InputFile> pathAndInputFiles = new HashMap<>();
+        fileSystem.inputFiles(mainFilePredicate).forEach(inputFile -> {
+            pathAndInputFiles.put(Path.of(inputFile.uri()).toString(), inputFile);
+        });
+
+        String analyzedResultsPath = sensorContext.config().get(ANALYSIS_RESULTS_FILE_PATH).orElse(null);
+        if (analyzedResultsPath != null) {
+            processAnalyzedResultsReport(sensorContext, pathAndInputFiles, analyzedResultsPath);
+        } else {
+            performLibraryCall(sensorContext, pathAndInputFiles);
+        }
```

Comment:
Shall we use the more monadic style here?
```suggestion
        sensorContext.config().get(ANALYSIS_RESULTS_FILE_PATH).ifPresentOrElse(
                (analyzedResultsPath) -> processAnalyzedResultsReport(sensorContext, pathAndInputFiles, analyzedResultsPath),
                () -> performLibraryCall(sensorContext, pathAndInputFiles)
        );

```